scipy/linalg/_decomp.py

```diff
@@ -280,6 +281,7 @@ def eig(a, b=None, left=False, right=True, overwrite_a=False,
     return w, vr
 
 
+@_xp_dispatch(['a', 'b'], 'linalg')
```

Comment:
This is all it takes to get the dispatching working as long as the array arguments are all consecutive at the beginning of the signature.

Currently, this assumes the xp-backend function has the same signature as the SciPy function, so it doesn't emit helpful errors if this is not the case and unsupported arguments are passed. There are plenty of ways to deal with this, but I'd rather not go too much deeper into this without initial feedback.